docs/quality_assurance.md

```diff
@@ -0,0 +1,169 @@
+| 2.08 | ICS-20 channel creation | `Scheduled` (ibc-go team) | `??` | `Future work` | `Scheduled` | 
+| 2.09 | ICS-20 transfer | `Scheduled` (ibc-go team) | `??` | `NA` | `Scheduled` | 
+| 2.10 | Changes in IBC-GO testing suite | `Scheduled` (ibc-go team) | `NA` | `Partial coverage` | `NA` | 
+
+### Integration with Cosmos SDK
+
+A prerequisite of the code review is to open a PR with all the [SDK changes](https://github.com/cosmos/cosmos-sdk/tree/interchain-security-rebase) needed by Interchain Security.
+
+| ID | Concern | Code Review | Unit Testing | Diff. testing | Testnet | 
+| -- | ------- | ----------- | ------------ | ------------- | ------- | 
+| 3.01 | Changes to staking module | `Scheduled` (sdk team) | `Partial coverage` <br /> see [unbonding_test.go](../x/ccv/provider/unbonding_test.go) <br />  redelegation and validator unbonding missing | `Partial coverage` | `Scheduled` | 
+| 3.02 | Changes to slashing module | `Scheduled` (sdk team) | `Done` <br /> see [TestValidatorDowntime](../x/ccv/consumer/keeper/keeper_test.go#L345) <br />  | `NA` | `Scheduled` | 
+| 3.03 | Changes to evidence module | `Scheduled` (sdk team) | `Done` <br /> see [TestValidatorDoubleSigning](../x/ccv/consumer/keeper/keeper_test.go#L427) <br />  | `NA` | `Scheduled` | 
+
+### Provider Chain Correctness
+
+The main concern addressed in this section is the correctness of the provider chain (e.g., the Cosmos Hub). In other words, when Interchain Security is enabled (i.e., the provider CCV module is enabled), the safety and liveness properties still hold. This _**MUST**_ be the case regardless of the number of consumer chains, i.e., 
+- no consumer chain;
+- one single consumer chain;
+- multiple consumer chains.
+
+| ID | Concern | Code Review | Unit Testing | Diff. testing | Testnet | Protocol audit |
+| -- | ------- | ----------- | ------------ | ------------- | ------- | ----- |
+| 4.01 | Liveness of undelegations <br /> - unbonding delegation entries are eventually removed from `UnbondingDelegation` | `Scheduled` | `Done` <br /> see [unbonding_test.go](../x/ccv/provider/unbonding_test.go) | `Done` | `Scheduled` | `Scheduled` |
+| 4.02 | Liveness of redelegations <br /> - redelegations entries are eventually removed from `Redelegations` | `Scheduled` | `Scheduled` | `Scheduled` | `Scheduled` | `Scheduled` |
+| 4.03 | Liveness of validator unbondings <br /> - unbonding validators with no delegations are eventually removed from `Validators` | `Scheduled` | `Scheduled` | `Done` | `Scheduled` | `Scheduled` |
+| 4.04 | Unbonding operations (undelegations, redelegations, validator unbondings) should eventually complete even if the CCV channel is never established (due to error) <br /> - expected outcome: the pending VSC packets eventually timeout, which leads to the consumer chain removal | `Scheduled` | `??` | `Future work` | `Scheduled` | `Scheduled` <br /> high priority |
```

Comment:
This is an interesting one. What's the longest possible unbonding delay here? A way to avoid this altogether would be to have an ack from the consumer chain that a validator update had been received and not stop unbonding until then. That would be a lot of work though

Reply:
> A way to avoid this altogether would be to have an ack from the consumer chain that a validator update had been received

How can a validator update be received without an established CCV channel?